Giftcard component

yassirog opened the pull request:
fixed #116 #121 #120 #124 #125 #129 #130 

## Wat is er veranderd?
### 🔧 Functionaliteit & Componenten

- Ik heb de giftcard-component opgezet en geïmplementeerd (#116).
- Ik heb een knop toegevoegd om giftcards te activeren (#121).
- Ik heb een formulier gemaakt met invoervelden voor de giftcard-pagina (#120).
- Ik heb contactinformatie toegevoegd aan de pagina (#124).
- Ik heb een bug opgelost waarbij giftcards niet als klikbare knoppen werkten (#125).

### 🗄️ Data & Logica

- Ik heb data vanuit de database toegevoegd om giftcards dynamisch te laden (#129).
- Ik heb een error state toegevoegd om foutmeldingen goed af te vangen (#130).
- Ik heb aria-pressed toegevoegd voor betere toegankelijkheid.

### 📦 Optimalisaties & UX

- Ik heb lazy loading, skeleton loading en local storage toegevoegd om de gebruikservaring te verbeteren.
- Tot slot heb ik een fix gedaan voor Netlify en ben ik begonnen met het responsief maken van de giftcard-pagina (#131).

<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->

## Hoe is dit getest?

- [x] User test
- [x] Accessibility test
- [x] Responsive Design test
- [ ] Device test
- [x] Browser test
- [x] Contrast test

### Testen
**User Test**
**Inputs te klein**
Ik heb voor de usertest Rukiya en Esha mijn pagina laten testen (giftcard page). Ik zag het zelf ook al, maar wat er fout was is dat mijn inputs veel te klein waren voor mobiel. Ik had mijn schets overgenomen, maar in mijn schetsen had ik de form inputs veelste klein gemaakt. Dit was beide hun feedback, maar ik had dit al verwacht. Ik liet Esha toen nieuwe schetsen maken en had dit ook geimplementeerd. Ik zal ook mijn en Esha's schetsen laten zien in de Images onderin.
**Giftcards waren geen buttons**
Later liet ik ook Mohammed een test doen, maar niet alleen op de website maar ik liet hem in mijn code duiken. Ik had een erg domme fout gemaakt, ik had de giftcards die als buttons dienen, geen button tag gegeven. Nadat hij dit aan me had verteld had ik het direct gefixt.

**Accessibility test**
Mijn pagina heeft bij de accessibility test 89 gescoord. Waarom is het geen 100? Nou precies omdat er fouten zitten in de footer i.p.v in mijn pagina zelf. De footer is geimplementeerd op elke pagina, en is niet mijn code. De fouten die er zijn staan ook weer in de Images onderin.

**Responsiveness test**
Aangezien ik alleen de mobiele versie af heb, kan ik niet echt de responsiveness testen, ik heb nu alleen kunnen kijken of het kapot gaat zodra ik de scherm erg klein maak en dat is niet zo. Dus dat is wel goede feedback.

**Browser test**
Ik heb de pagina op verschillende browsers getest waarbij ik op alle schermen dezelfde resultaat kreeg. Ik heb hier nauwkeurig opgelet en wou dat het voor WOGO toegankelijk was voor elke browser. Ik wou niet dat het anders eruit zou zien dan op andere browsers. Ik heb hierdoor dus ook een scroll animatie verwijdert die op sommige browsers wel liet zien en op andere dus niet.

**Contrast test**
Ik heb ook de contrast getest op de button, andere dingen contrast heb ik niet getest, omdat ik die in de vorige pullrequest al had getest en wist dus dat die goed waren. Ik heb dus de button getest en deze scoord een 11.31 (foto hiervan is ook in Images onderin).

## Images
**_Mijn schets_**
![image](https://github.com/user-attachments/assets/ea78e0ef-1f40-4250-bc39-29b6ee216957)

_**Esha's schets**_
![image](https://github.com/user-attachments/assets/ddb995a8-2467-4955-8ee7-10be290c4eb7)

_**Accessibility test**_
![image](https://github.com/user-attachments/assets/5688cc5f-67c6-4440-867a-0b38140f96fd)

_**Contrast test**_
![image](https://github.com/user-attachments/assets/4a074af8-a46c-41f3-87e1-6b5c3dd7492f)


<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->



<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->
## How to review
- Check if there are mistakes in code
- Check if code is working

RukiyaTossou committed:
Merge pull request #98 from fdnd-agency/dev

Dev

yassirog committed:
✨ FEAT: giftcard component, fixes #116

fixed #116

- Made a component for a giftcard as the design made in Figma

yassirog reviewed (commented):
Looks good - Jesse

kwinsie commented:
### Design 
- Maybe use some more colours from the website
- Give the elements a bit more space
- Maybe it would be nice to use multiple different screens in the process to buy a giftcard

lemuelmgsn commented:
De cadeaukaartjes zien er goed en leuk uit!

Misschien kan je de cadeaukaartjes wat meer versieren door een soort silhouette toe te voegen van cocktail glazen.

![image](https://github.com/user-attachments/assets/0840a189-b3ea-4166-9224-ee6a5ccb9c01)

![image](https://github.com/user-attachments/assets/0a376b4d-ddac-4076-a837-6be7b8a539a3)

yassirog committed:
FEAT: giftcard button added, fixes #121 ✨

fixed #121

- added a giftcard button & component

yassirog committed:
feat: form with inputs for giftcard-page #120 ✨

fixes #120

- made new component for the form
- made form with inputs to buy a giftcard

yassirog committed:
feat: added contact information, fixes #124 ✨

fixed #124

- Added a new section with the contact information

yassirog committed:
bug: giftcards are now buttons, fixes #125

fixed #125

- Giftcards are now buttons instead of articles. This way you can select them.

yassirog committed:
FEAT: added database data for loading giftcards, fixes #129

fixed #129

- Added code to load data from the contentful database for the giftcards

yassirog committed:
feat: aria-pressed added

- Added aria-pressed for better accessibility for screenreaders.

yassirog committed:
feat: added error state, fixes #130

fixed #130

- Added a error state on the giftcard page

yassirog committed:
added: lazy loading, skeleton loading, local storage to giftcard page

added few new things to the giftcard page

yassirog committed:
feat: fixed netlify & started responsiveness on giftcard page, fixes #131

- Changed variabele in server.js for netlify
- Made components on website to start adjusting responsiveness for the giftcard page
